### DISCUSSION TITLE
Does govips support streaming?

### DISCUSSION THREAD

**Discussion by enricoribelli** [davidbyttow/govips#401] [Q&A] [answered]
Hello team!
I am looking for some docs stating clearly (and eventually with examples) that govips supports stream of images.
I currently need to process big images and therefore I don't want to load everything in memory but stream in and stream out.
libvips currently supports this feature as explained [here](https://www.libvips.org/2019/11/29/True-streaming-for-libvips.html) . However I was diving into the govips code to find references to [new_from_descriptor](https://www.libvips.org/API/current/VipsTargetCustom.html#vips-source-new-from-descriptor:~:text=from%20a%20pipe-,vips_source_new_from_descriptor,-()) as pointed out from that documentation. The closest thing I could find to a stream is [NewImageFromReader](https://pkg.go.dev/github.com/davidbyttow/govips/v2/vips#NewImageFromReader:~:text=func%20NewImageFromReader%20%C2%B6) but which calls [NewImageFromBuffer](https://pkg.go.dev/github.com/davidbyttow/govips/v2/vips#NewImageFromBuffer:~:text=func%20NewImageFromBuffer%20%C2%B6) and I don't think this reads anything from a stream since it suggests the load of the image in a buffer. Could someone point me to the right direction by any chance? And is there somewhere clear docs that I am missing explaining a bit deeper what does what or only way is yolo-ing in the code?

**Comment by tonimelisma** [comment, marked as answer]
Hey @enricoribelli, no it doesn't as of now. There's been a couple of attempts to add streaming support that you can find here: https://github.com/davidbyttow/govips/issues?q=streaming

Any PRs would gladly be merged but I fear that it's not a trivial implementation.

**Reply by enricoribelli** [reply]
Thanks for your reply!
